src/Node/Render.php

```diff
@@ -100,7 +144,13 @@ public function getVariantContext($variant, $config) {
   }
 
   /**
-   * Returns the sanitized component name
+   * Returns the sanitized component name without its variant.
+   *
+   * @param string $component
+   *   The component name to sanitize.
+   *
+   * @return string
+   *   The base component name without its variant.
    */
   public function sanitizeComponentName($component) {
```

Comment:
Architecturally, it does not make sense to execute this method more than once.  Also, "sanitize" is a misleading name, the term is only used for cleaning/escaping operations.  The operation performed here is much rather "extract".

Can we determine this information directly in the constructor and set the result to properties?

```php
$this->variants = explode('--', $component);
$this->component = array_shift($this->variants);
```

That's essentially what `getComponentParts()` is doing, but supports multiple variants.  If you prefer to keep a separate method for unit testing purposes, then I suggest to make it
```
return [$component, $variants];
```
and perform the assignment to class properties in the calling code (constructor).